pkg/lib/packagekit.js

```diff
@@ -452,7 +454,7 @@ export function check_missing_packages(names, progress_cb) {
         }
     }
 
-    return refresh().then(resolve)
+    return (skip_refresh ? resolve() : refresh().then(resolve))
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19281-20230919-044351-80b8161f-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)